app/models/requests.go

```diff
@@ -189,30 +188,41 @@ type SendMessageRequest struct {
 	QueueUrl                string                      `json:"QueueUrl" schema:"QueueUrl"`
 }
 
-func (r *SendMessageRequest) SetAttributesFromForm(values url.Values) {
+func parseMessageAttributes(values url.Values, keyPrefix string) map[string]MessageAttribute {
```

Comment:
Could we have a couple tests on this directly?  I think the smoke tests cover a lot of this, you're right, but they serve a bit of a different purpose.  You can cannibalize the tests for `SetAttributesFromForm` if you want to (if you want to, you don't have to if that's too much), and going forward, I won't have to write tests for all this logic in each case.

Reply:
Done!